controllers/machinedeletionremediation_controller_test.go

```diff
@@ -165,19 +171,36 @@ var _ = Describe("Machine Deletion Remediation CR", func() {
 					verifyMachineIsDeleted(workerNodeMachineName)
 					verifyMachineNotDeleted(masterNodeMachineName)
 
-					// The transition of the Processing Condition status from
-					// unset to "Started", and finally to "Finished" is too
-					// fast to test the initial value ("Started") by inspecting
-					// the actual MDR CR. For this reason the initial value is not
-					// tested here.
+					// Machine is deleted, but the remediation is not completed yet
+					verifyConditionsMatch([]expectedCondition{
+						{commonconditions.ProcessingType, metav1.ConditionTrue, remediationStarted},
+						{commonconditions.SucceededType, metav1.ConditionUnknown, remediationStarted},
+						// Cluster provider is not set in this test
+						{commonconditions.PermanentNodeDeletionExpectedType, metav1.ConditionUnknown, v1alpha1.MachineDeletionOnUndefinedProviderReason}})
+
+					// Mock Machine and Node re-provisioning (even though this test does not actually delete the node, just the machine).
+					// 1. Create a Machine's replacement with a new name
+					// 2. Update WorkerNode's annotation to point to the new Machine
+					machineReplacementName := workerNodeMachineName + "-replacement"
+					workerNode.Annotations[machineAnnotationOpenshift] = fmt.Sprintf("%s/%s", defaultNamespace, machineReplacementName)
+					workerNodeMachineReplacement := createWorkerMachine(machineReplacementName)
+					Expect(k8sClient.Update(context.Background(), workerNode)).To(Succeed())
+					DeferCleanup(k8sClient.Delete, workerNodeMachineReplacement)
+					Expect(k8sClient.Create(context.Background(), workerNodeMachineReplacement)).To(Succeed())
```

Comment:
I think these lines can be logically reordered better:

1. Update `workerNode`
2. Create object `createWorkerMachine` 
3. Try to create the object `workerNodeMachineReplacement` in the cluster
4. DeferCleanup the object

Reply:
(step 4 can also be before step 3)

Reply:
> (step 4 can also be before step 3)

Yes, I think `DeferCleanup` always precedes creation

For the other points, I'm not sure the current flow is less logical. What do you see in this order that doesn't click right?

My idea in this case was to create first the new `WorkerNodeMachineReplacement`, then change the reference in the `WorkerNode` (which is kind of orphan at this point, since its `WorkerNodeMachine` was deleted). I know that the reference is just a string, and the Machine doesn't need to exist already, but I think that is logical as well.

Reply:
> What do you see in this order that doesn't click right?

I think I understood it wrong, as the status update is unrelated to the creation of the new machine.
If they are unrelated, then the update can be done prior to the creation of `WorkerNodeMachineReplacement` and it should be "grouped" with the [two lines beforehand](https://github.com/medik8s/machine-deletion-remediation/pull/92/files#diff-867a8834813cdf408e2387961fc75a31146379802d5abec6f465f1d8a565b903R184-R185). Because `WorkerNodeMachineReplacement`  can be created after we check if the update (to workerNode) has been succeeded.

But since the update and machine creation are related, then it makes sense to update after the machine has been created.

>the Machine doesn't need to exist already

The test should be reflect (or be close) to reality, thus if the machine must be existed prior to updating `WorkerNode`, then I am thinking that the update should be done after creating the machine. Otherwise, you may ignore my (long) suggestion

Reply:
> thus if the machine must be existed prior to updating WorkerNode, then I am thinking that the update should be done after creating the machine

I understood now, yes that'd be better :+1:

Reply:
"Much ado about nothing" :smile: 

You're still right and the commented code is still needed, but we also need `VerifyConditionMaches` to wait up to 60s, because MDR checks node count restoration every 30s, so a `VerifyConditionMaches`'s timeout of 20s is not enough (most of the times) :smile: